src/test/java/org/folio/rest/impl/OaiPmhImplTest.java

```diff
@@ -2457,27 +2485,38 @@ void verifyResumptionTokenFlow_whenVerbListRecordsAndMetadataPrefixMarc21WithHol
   }
 
   private ResumptionTokenType makeResumptionTokenRequestsAndVerifyCount(List<HeaderType> totalRecords,
-                                                                        ResumptionTokenType resumptionToken, int desiredCount, int expectedCursor) {
+      ResumptionTokenType resumptionToken, VerbType verb, int desiredCount, int expectedCursor) {
     RequestSpecification request;
     OAIPMH oaipmh;
     List<HeaderType> records;
-    request = createBaseRequest()
-      .with()
-      .param(VERB_PARAM, LIST_RECORDS.value())
+    request = createBaseRequest().with()
+      .param(VERB_PARAM, verb.value())
       .param(RESUMPTION_TOKEN_PARAM, resumptionToken.getValue());
-    oaipmh = verify200WithXml(request, LIST_RECORDS);
-    verifyListResponse(oaipmh, LIST_RECORDS, desiredCount);
-    resumptionToken = getResumptionToken(oaipmh, LIST_RECORDS);
+    oaipmh = verify200WithXml(request, verb);
+    verifyListResponse(oaipmh, verb, desiredCount);
+    resumptionToken = getResumptionToken(oaipmh, verb);
     assertThat(resumptionToken, is(notNullValue()));
-    assertEquals(expectedCursor, resumptionToken.getCursor().intValue());
+    assertEquals(expectedCursor, resumptionToken.getCursor()
+      .intValue());
 
-    records = oaipmh.getListRecords().getRecords().stream()
-      .map(RecordType::getHeader)
-      .collect(Collectors.toList());
+    records = getHeadersListDependOnVerbType(verb, oaipmh);
     totalRecords.addAll(records);
     return resumptionToken;
   }
 
+  private List<HeaderType> getHeadersListDependOnVerbType(VerbType verb, OAIPMH oaipmh) {
+    if (verb.equals(LIST_RECORDS)) {
+      return oaipmh.getListRecords()
+        .getRecords()
+        .stream()
+        .map(RecordType::getHeader)
+        .collect(Collectors.toList());
+    } else {
+      return oaipmh.getListIdentifiers()
+        .getHeaders();
+    }
```

Comment:
I'd use ternary operation here, what do you think?

Reply:
Good suggestion